src/server/tran_server.cpp

```diff
@@ -341,13 +342,18 @@ tran_server::disconnect_all_page_servers ()
 {
   assert_is_tran_server ();
 
-  for (const auto &conn : m_page_server_conn_vec)
-    {
-      er_log_debug (ARG_FILE_LINE, "Transaction server disconnected from page server with channel id: %s.\n",
-		    conn->get_channel_id ().c_str ());
-      conn->disconnect ();
-    }
-  m_page_server_conn_vec.clear ();
+  // We assumes that no threads are waiting for response at this moment so that m_conn->stop_response_broker() is not needed.
```

Comment:
Can we really assume it? I assume it because all outgoing msg generators stop and all transactions aborts before disconnecting connections. However, I'm not sure they escape from blocking on `send_receive()`..

Reply:
> I think we will have to add some timeouts in a few places.

I'll create the issue if you don't mind, or you do it if you have a more detailed image.

> it crashed, as expected

You mean the crash because there is no connection in ATS by the added assertion, right? I ran into it, but I think it's just undefined behavior. But, yes, the PS should get out of the hang at least. I'll look into it.

Reply:
> Can we really assume it? I assume it because all outgoing msg generators stop and all transactions abort before disconnecting connections (`ts_Gl->stop_outgoing_page_server_messages` and `log_abort_all_active_transaction()`) . However, I'm not sure they have a way out from blocking on `send_receive()`.

http://jira.cubrid.org/browse/LETS-687